.github/workflows/docker.yml

```diff
@@ -0,0 +1,33 @@
+name: docker
+
+on:
+    push:
+        branches:
+            - master
+
+jobs:
+    build-docker-image:
+        runs-on: ubuntu-18.04
+        steps:
+            - uses: actions/checkout@master
+
+            - name: Dockerhub login
+              env:
+                  DOCKER_PASSWORD: ${{ secrets.DOCKER_PASSWORD }}
+                  DOCKER_USERNAME: ${{ secrets.DOCKER_USERNAME }}
+              run: |
+                  echo "${DOCKER_PASSWORD}" | docker login --username ${DOCKER_USERNAME} --password-stdin
+
+            - name: Set up Docker Buildx
+              id: buildx
+              uses: crazy-max/ghaction-docker-buildx@v1
```

Comment:
Do you think it's possible to avoid this external dep and [build buildx like below](https://github.com/docker/buildx#building)?

```
$ git clone git://github.com/docker/buildx && cd buildx
$ make install
```

Reply:
There were a couple extra steps, but think I have it now. I'm running it locally, so it'll take some time to finish, but looks like it's working.

Reply:
This worked for me locally, is there a way to test it out on your Github server before merging?

Reply:
> There were a couple extra steps, but think I have it now. I'm running it locally, so it'll take some time to finish, but looks like it's working.

Nice! I'm not sure how to test it on our server, but if it works on yours then I assume it'll work on ours. I might just merge it and give it a go! :)